### ISSUE TITLE
Support ShellCheck `source-path` directive

### ISSUE THREAD

**Issue by Bilge** [BashSupport-Pro/bashsupport-pro#141]
Although BashSupport Pro appears to [support](https://www.bashsupport.com/manual/navigation/sourced-files/#shellcheck-support) `# shellcheck source` directives, it does not support [`source-path`](https://github.com/koalaman/shellcheck/wiki/Directive#source-path). This means we have to add every single source file with a `source` directive, since  `source-path` is not available.

**Comment by jansorg** [comment]
You‘re right, that‘s currently unsupported and a limitation. I‘ll look into this and how to support it…

**Comment by jansorg** [comment]
The next beta of version 4 will support `source-path=`.

As first step, it's going to support the following cases:
```bash
#!/bin/bash
# shellcheck source-path=global/search/path

# shellcheck source=SCRIPTDIR/file-inside-global-search-path.bash
. file-inside-global-search-path.bash

# shellcheck source-path=additional-search-path;another-search-path
. some-file-in-any-search-path.bash
```

The next steps will include support for code completion for values of `source-path` and completion based on search-path for `source=` directives. But I currently can't promise if that's going to be available with 4.0.

Do you have any particular use-case which has to be supported by `source-path`?

**Comment by jansorg** [comment]
4.0 is available now.
